### PR TITLE
fix(site): add workspace proxy section to health page

### PR DESCRIPTION
**Pull request by johnstcn**
- Adds workspace proxy section to health page
- Conditionally places workspace proxy warnings in errors or warnings based on calculated severity
- Adds some more stories we were missing for HealthPage

### CODE REVIEW

File: site/src/pages/HealthPage/HealthPage.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const sections = {
   access_url: "Access URL",
   websocket: "Websocket",
   database: "Database",
+  workspace_proxy: "Workspace Proxy",
```
Comment (mtojek) [inline site/src/pages/HealthPage/HealthPage.tsx:32]: nit: You could add something to the storybook too.

File: coderd/healthcheck/workspaceproxy.go [hunk 2]
```diff
@@ -78,6 +79,7 @@ func (r *WorkspaceProxyReport) Run(ctx context.Context, opts *WorkspaceProxyRepo
 	})
 
 	var total, healthy int
+	errs := make([]string, 0)
```
Comment (mtojek) [inline coderd/healthcheck/workspaceproxy.go:82]: nit: I never know which form is better. Yours or `var errs []string`.

**Review by mtojek — APPROVED** [review]
👍

File: coderd/healthcheck/workspaceproxy.go (hunk 2 above)
Comment (johnstcn) [reply coderd/healthcheck/workspaceproxy.go:82]: I suppose this form is better if you know exactly how many elements there are to be in the slice, which we don't.

**Pull request merged**